docs/notebooks/mixed_search_spaces.pct.py

```diff
@@ -236,35 +236,41 @@
 model = GaussianProcessRegression(gpflow_model)
 
 # %% [markdown]
-# We create a trust region acquisition rule that uses the efficient global optimization (EGO)
-# acquisition rule as the base rule. The trust region acquisition rule is initialized with a set of
+# We create a trust region meta acquisition rule that uses efficient global optimization (EGO)
+# as the base rule. The trust region rule is initialized with a set of
 # trust regions; 5 in this example. Each trust regions is defined as a product of a discrete and a
-# continuous trust sub-region. The base rule is then used to optimize the acquisition function
-# within each trust region. This setup is similar to the one used in the "Batch trust region rule"
-# section of the [trust region Bayesian optimization notebook](trust_region.ipynb).
+# continuous trust sub-region, analogous to a `TaggedProductSearchSpace`. The base rule is then
+# called to optimize the acquisition function within each region.
 #
-# Analogous to a `TaggedProductSearchSpace`, each trust region is a product of a discrete and a
-# continuous trust (sub-)region. The discrete sub-region is defined by
-# `FixedPointTrustRegionDiscrete` that selects a random point from the discrete space, which is then
-# fixed for the duration of the optimization. The continuous sub-region is defined by
-# `SingleObjectiveTrustRegionBox`, just as in the trust region notebook, where the region is updated
-# at each step of the optimization.
+# This setup is similar to the one used in the "Batch trust region rule"
+# section of the [trust region Bayesian optimization notebook](trust_region.ipynb). That notebook
+# creates trust regions of type `SingleObjectiveTrustRegionBox`. Here, we create trust regions that
+# are a product of a discrete and a continuous trust sub-region with
+# `UpdatableTrustRegionProduct`. The continuous part `SingleObjectiveTrustRegionBox` is the same as
```

Comment:
The text is good, but I am wondering (prob not for this PR) if it would be better to do this automatically? As in, it is probably a bit of a headache for the use to know that if they have a mixed input space then they must use a TrustRegionProduct. Ideally they would use SingleObjectiveTRBox and we dispatch it automatically to the right kind?

Reply:
I'll think about this for a separate PR.